rust/parquet/src/schema/types.rs

```diff
@@ -972,18 +1011,22 @@ fn from_thrift_helper(
 }
 
 /// Method to convert to Thrift.
-pub fn to_thrift(schema: &Type) -> Result<Vec<SchemaElement>> {
+pub fn to_thrift(schema: &Type, writer_version: i32) -> Result<Vec<SchemaElement>> {
```

Comment:
Can you elaborate a bit why `writer_version` is needed? I thought it would just be a straightforward conversion from the `SchemaElement` to the thrift counterpart.

Reply:
I'll add a comment if you agree with my logic below.

I understand the format to mean that `LogicalType` is a version 2 only detail, such that someone writing version 1 of the format, would not expect a `LogicalType` to be populated. So, I'm checking if one is intending on writing v2, and only populating `LogicalType` in that instance.

This would become relevant when parsing the schema and displaying it (future PR). a v2 file's text schema has `INTEGER(32,true)` instead of `INT_32`, so to ensure that this is the case, we shouldn't write the `LogicalType` for v1 files; as a loose reader could end up misinterpreting the schema.

Reply:
Hmm, the `LogicalType` is not version 2 only though: it was introduced in 2.4.0. It is also an optional field which means backward compatibility. Therefore, I guess it should be fine to write it no matter the `writer_version` is 1 or 2? A reader > 2.4.0 will try to parse the optional `LogicalType` first while one < 2.4.0 will just look at the `ConvertedType` field.

I don't fully understand the implication of parsing schema. When looking at `INTEGER(32,true)`, shouldn't we just populate the `LogicalType` together with the `ConvertedType` field? and only populate `ConvertedType` when seeing `INT_32`?

Reply:
I see what you mean @sunchao. I've removed the version check, and always write the logical type. I suppose I'm not thinking of this well from a compatibility perspective. We'll always want to write complying with whatever `parquet-format` version we're using.

There's still something that's unclear to me about how we'll deal with the text schema format, but I can raise the questions when I work on its relevant PR.